common/src/main/java/io/netty/util/internal/logging/Slf4JLoggerFactory.java

```diff
@@ -40,31 +37,8 @@ public Slf4JLoggerFactory() {
 
     Slf4JLoggerFactory(boolean failIfNOP) {
         assert failIfNOP; // Should be always called with true.
-
-        // SFL4J writes it error messages to System.err. Capture them so that the user does not see such a message on
```

Comment:
It was really long time ago I wrote this, so I'm not completely sure.

Once upon a time, SLF4J was not a dominant logging framework, lots of people were still using commons-logging or log4j while some others use SLF4J. As a result, our classpath had slf4j-api.jar, but not the implementation JAR (e.g. slf4j-log4j.jar). If Netty chose SLF4J just because slf4j-api.jar is in the classpath, nothing will be logged at all because NOPLoggerFactory would be used.

Therefore, when the current LoggerFactory is NOPLoggerFactory, Slf4JLoggerFactory raised NoClassDefException so that Netty tries other logging framework automatically. However, during this process, SLF4J will write some warning/error messages to System.err. I thought this information is not very useful to users who expects Netty to use other logging framework than SLF4J, so I decided to hide it.

But it alters the global resource (System.err) which is obviously not a good idea. Please feel free to remove it.